include/ignition/physics/World.hh

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#ifndef IGNITION_PHYSICS_WORLD_HH_
+#define IGNITION_PHYSICS_WORLD_HH_
+
+#include <string>
+
+#include <ignition/physics/FeatureList.hh>
+
+namespace ignition
+{
+  namespace physics
+  {
+    /////////////////////////////////////////////////
+    class IGNITION_PHYSICS_VISIBLE PhysicsOptions : public virtual Feature
+    {
+      /// \brief The World API for setting physics options.
+      public: template <typename PolicyT, typename FeaturesT>
+      class World : public virtual Feature::World<PolicyT, FeaturesT>
+      {
+        /// \brief Set the name of the collision detector to use.
+        /// \param[in] _collisionDetector Name of collision detector.
+        public: void SetCollisionDetector(
```

Comment:
Since setting a collision detector may not be possible for other engines, how about splitting this into two features?

Reply:
Done ee59d1a